execution_engine/src/storage/trie_store/operations/mod.rs

```diff
@@ -687,7 +689,12 @@ where
         let path_to_node: Vec<u8> = get_parents_path(&parents);
         // Check that the path to the node is a prefix of the current path
         let current_path = common_prefix(path_to_leaf, &path_to_node);
-        assert_eq!(current_path, path_to_node);
+        assert_eq!(
+            current_path, path_to_node,
+            "current_path must be the same as path_to_node"
+        );
+        error!("current_path must be the same as path_to_node");
```

Comment:
This will be printed even when `current_path == path_to_node`.  We need to do something more like:
```suggestion
        if current_path != path_to_node {
            error!(?current_path, ?path_to_node, "paths should be equal");
            debug_assert!(
                false,
                "current path {:?} != path to node {:?}",
                current_path, path_to_node
            );
        }
```